src/providers/idp/idp_id.c

```diff
@@ -0,0 +1,675 @@
+        DEBUG(SSSDBG_OP_FAILURE,
+              "Failed to allocate memory for IdP request.\n");
+        return ENOMEM;
+    }
+
+    state->idp_req->idp_options = state->idp_id_ctx->idp_options;
+
+    ret = set_oidc_extra_args(state, state->idp_id_ctx, state->lookup_type,
+                              state->filter_value,
+                              state->filter_type,
+                              state->extra_value,
+                              &state->idp_req->oidc_child_extra_args);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_OP_FAILURE, "set_oidc_extra_args() failed.\n");
+        return ret;
+    }
+
+    state->send_buffer = talloc_zero(state, struct io_buffer);
+    if (state->send_buffer == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "talloc_zero failed.\n");
+        return ENOMEM;
+    }
+
+    state->send_buffer->size = strlen(state->idp_id_ctx->client_secret);
+    state->send_buffer->data = talloc_memdup(state->send_buffer,
+                                             state->idp_id_ctx->client_secret,
+                                             state->send_buffer->size);
+    if (state->send_buffer->data == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "Failed to create send buffer.\n");
+        return ENOMEM;
+    }
+
+    subreq = handle_oidc_child_send(state, state->ev, state->idp_req,
+                                    state->send_buffer);
+    if (subreq == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "handle_oidc_child_send() failed.\n");
+        return ENOMEM;
+    }
+    tevent_req_set_callback(subreq, idp_type_get_done, req);
+
+    return ret;
+}
+
+static void idp_type_get_done(struct tevent_req *subreq)
+{
+    struct idp_type_get_state *state;
+    struct tevent_req *req;
+    errno_t ret;
+
+    uint8_t *buf;
+    ssize_t buflen;
+
+    req = tevent_req_callback_data(subreq, struct tevent_req);
+    state = tevent_req_data(req, struct idp_type_get_state);
+
+    ret = handle_oidc_child_recv(subreq, state, &buf, &buflen);
+    talloc_zfree(subreq);
+    if (ret != EOK) {
+        state->dp_error = DP_ERR_FATAL;
+        tevent_req_error(req, ret);
+        return;
+    }
+
+    DEBUG(SSSDBG_TRACE_ALL, "[%zd][%.*s]\n", buflen, (int) buflen, buf);
+    switch (state->lookup_type) {
+    case IDP_LOOKUP_USER:
+        ret = eval_user_buf(state->idp_id_ctx, NULL, buf, buflen);
+        break;
+    case IDP_LOOKUP_GROUP:
+        ret = eval_group_buf(state->idp_id_ctx, NULL, buf, buflen);
+        if (ret == EOK) {
+            DEBUG(SSSDBG_TRACE_ALL, "Looking up group members.\n");
+
+            state->lookup_type = IDP_LOOKUP_GROUP_MEMBERS;
+            talloc_zfree(state->idp_req);
+
+            ret = idp_type_get_step(req);
+            if (ret == EOK) {
+                return;
+            }
+            DEBUG(SSSDBG_OP_FAILURE, "Failed to lookup group members.\n");
+        }
+        break;
+    case IDP_LOOKUP_GROUP_MEMBERS:
+        ret = eval_user_buf(state->idp_id_ctx, state->filter_value,
```

Comment:
Shouldn't `state->noexist_delete` be propagated to and used in `eval_*()` functions?

Reply:
Added together with the deletion of not existing users or groups.